search-app/test/cmr/search/site/data_test.clj

```diff
@@ -77,9 +80,33 @@
      :stac-docs-url "https://cmr.uat.earthdata.nasa.gov/stac/docs/index.html"
      :static-cloudstac-url "https://cmr.uat.earthdata.nasa.gov/static-cloudstac"}
 
-    ;; localhost 
+    ;; localhost
     "http://localhost:3003"
-    {:stac-url "http://localhost:3000/stac" 
+    {:stac-url "http://localhost:3000/stac"
      :cloudstac-url "http://localhost:3000/cloudstac"
      :stac-docs-url "http://localhost:3000/stac/docs"
      :static-cloudstac-url "http://localhost:3000/static-cloudstac"}))
+
+(deftest client-id-tests
+  (testing "check for client id from inside of endpoint-get"
+    ;; Construct a funky context to trigger the special branch inside of get-providers
```

Comment:
what is 'special branch'?

Reply:
:execution-context is not a "standard" item in a context, this is used to trigger a defmethod. For all my unit tests I build contexts by hand instead of receiving one like you would in a system-int-test. So these tests end up being more aware of what goes on to trigger a test use case.

Reply:
I see. Okay that makes sense